File: binsV1/binv10/third_problem.rb
require 'json'

n = ARGV[1].to_i
n = 2 if n.zero?

products = [
  { "name": "Apples", "quantity_sold": 120 },
  { "name": "Oranges", "quantity_sold": 75 },
  { "name": "Bananas", "quantity_sold": 200 }
]

sortedProducts = products.sort_by { |data| data[:quantity_sold] }.reverse

highestSoldProducts = sortedProducts.take(n)

puts JSON.pretty_generate({ data: highestSoldProducts })

File: binV8/third_problem.js
const n = Number(process.argv[2]) || 2;

products = [
  { name: "Apples", quantity_sold: 120 },
  { name: "Oranges", quantity_sold: 75 },
  { name: "Bananas", quantity_sold: 200 },
];

const sortedProducts = products.sort(
  (a, b) => b.quantity_sold - a.quantity_sold
);

const highestSoldProducts = sortedProducts.slice(0, n);

console.log(JSON.stringify({ data: highestSoldProducts }, null, 2));
